report/src/main/webapp/app/services/history.service.js

```diff
@@ -0,0 +1,136 @@
+  'use strict';
+  angularAMD.factory('historyService', historyService);
+
+  /**
+   * Service responsible for fetching suite's history
+   */
+  function historyService($rootScope, $http, endpointConfiguration) {
+    var suiteHeaders;
+    $rootScope.endpointUrl = endpointConfiguration.getEndpoint().getUrl;
+    var service = {
+      getNextVersion: getNextVersion,
+      getPreviousVersion: getPreviousVersion,
+      fetchHistory: fetchHistory,
+    };
+
+    return service;
+
+    function fetchHistory(currentVersion, fetchCallback) {
+      $rootScope.selectedVersion = currentVersion;
+      getSuiteHistory(suiteHeaders, $rootScope, $http, function () {
+        fetchCallback();
+      });
+    }
+
+    function buildApiPath($allParametersList) {
+      return $rootScope.endpointUrl + 'history' + '?' +
+        $allParametersList[0] + '&' +
+        $allParametersList[1] + '&' +
+        $allParametersList[2];
+    }
+
+    function getPreviousVersion(currentVersion) {
+      var prevVersion = null;
+      $rootScope.suiteHeaders.forEach(function(suiteHeader, index) {
+        if(suiteHeader.version === currentVersion) {
+          if($rootScope.suiteHeaders[index + 1]) {
+            prevVersion =  $rootScope.suiteHeaders[index + 1].version;
+          }
+        }
+      });
+      return prevVersion;
+    }
+
+    function getNextVersion(currentVersion) {
+      var nextVersion = null;
+      $rootScope.suiteHeaders.forEach(function(suiteHeader, index) {
+        if(suiteHeader.version === currentVersion) {
+          if($rootScope.suiteHeaders[index - 1]) {
+            nextVersion =  $rootScope.suiteHeaders[index - 1].version;
+          }
+        }
+      });
+      return nextVersion;
+    }
+
+    function getSuiteHistory(suiteHeaders, $rootScope, $http, historyCallback) {
+      $rootScope.data = 'test';
+      var cUrl = new URL(window.location.href);
+      var company = cUrl.searchParams.get('company');
+      var project = cUrl.searchParams.get('project');
+      var suite = cUrl.searchParams.get('suite');
+      if (suite === null) {
+        var correlationId = cUrl.searchParams.get('correlationId');
+        suite = correlationId.split('-')[2];
+      }
+      var allParametersList = ['company=' + company, 'project=' + project, 'suite=' + suite];
+      return $http({
+        method: 'GET',
+        url: buildApiPath(allParametersList),
+        headers: {
+          'Content-Type': 'text/plain'
+        }
+      }).then(function (response) {
+        $rootScope.data = response.data;
+        suiteHeaders = response.data;
+        for (var i = 0; i < suiteHeaders.length; i++) {
+          var version = suiteHeaders[i].version;
+          var company = suiteHeaders[i].correlationId.split('-')[0];
+          var project = suiteHeaders[i].correlationId.split('-')[1];
+          var suite = suiteHeaders[i].correlationId.split('-')[2];
+          var correlationId = suiteHeaders[i].correlationId.split('-')[3];
```

Comment:
I think this you're parsing timestamp from the correlationID here, so this var should be renamed to e.g. `timestamp`
And as above - we should handle cases where company/project/suite have '-' characters